### PR TITLE
[Filesystem][Finder][Form] Use CPP

### PR DESCRIPTION
**Pull request by OskarStark**
| Q             | A
| ------------- | ---
| Branch?       | 7.1
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | --
| License       | MIT

### CODE REVIEW

File: src/Symfony/Component/Form/FormError.php [hunk 1]
```diff
@@ -20,13 +20,6 @@
  */
 class FormError
 {
-    protected string $messageTemplate;
```
Comment (xabbuh) [inline src/Symfony/Component/Form/FormError.php:23]: needs to be reverted

**Review by xabbuh — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Form/Exception/TransformationFailedException.php [hunk 2]
```diff
@@ -18,11 +18,13 @@
...
-    private array $invalidMessageParameters;
-
-    public function __construct(string $message = '', int $code = 0, \Throwable $previous = null, string $invalidMessage = null, array $invalidMessageParameters = [])
-    {
+    public function __construct(
+        string $message = '',
+        int $code = 0,
+        \Throwable $previous = null,
+        private ?string $invalidMessage = null,
+        private array $invalidMessageParameters = [],
```
Comment (xabbuh) [inline src/Symfony/Component/Form/Exception/TransformationFailedException.php]: I would revert the two lines above as the properties are actually initialised in `setInvalidMessage()`.

File: src/Symfony/Component/Form/FormError.php (hunk 1 above)
Comment (OskarStark) [reply src/Symfony/Component/Form/FormError.php:23]: done

File: src/Symfony/Component/HttpClient/Response/MockResponse.php [hunk 3]
```diff
@@ -43,9 +42,10 @@ class MockResponse implements ResponseInterface, StreamableInterface
      *
      * @see ResponseInterface::getInfo() for possible info, e.g. "response_headers"
      */
-    public function __construct(string|iterable $body = '', array $info = [])
-    {
-        $this->body = $body;
+    public function __construct(
+        private null|string|iterable $body = '',
```
Comment (xabbuh) [inline src/Symfony/Component/HttpClient/Response/MockResponse.php:46]:
```suggestion
        private string|iterable $body = '',
```

Comment (xabbuh) [inline src/Symfony/Component/HttpClient/Response/MockResponse.php]: I would revert this change. The possibility for `$body` to be `null` is purely internal. I guess we shouldn't allow consumers of this class to pass `null` here.

File: src/Symfony/Component/HttpClient/Retry/GenericRetryStrategy.php [hunk 4]
```diff
@@ -36,42 +36,35 @@ class GenericRetryStrategy implements RetryStrategyInterface
...
      * @param int   $maxDelayMs  Maximum delay to allow (0 means no maximum)
      * @param float $jitter      Probability of randomness int delay (0 = none, 1 = 100% random)
      */
-    public function __construct(array $statusCodes = self::DEFAULT_RETRY_STATUS_CODES, int $delayMs = 1000, float $multiplier = 2.0, int $maxDelayMs = 0, float $jitter = 0.1)
-    {
-        $this->statusCodes = $statusCodes;
-
+    public function __construct(
+        private array $statusCodes = self::DEFAULT_RETRY_STATUS_CODES,
+        private int $delayMs = 1000,
```
Comment (xabbuh) [inline src/Symfony/Component/HttpClient/Retry/GenericRetryStrategy.php]: I would revert this change as well. In other places where we check the passed value to (for example) throw exceptions we don't use CPP.

File: src/Symfony/Component/Form/FormConfigBuilder.php [hunk 5]
```diff
@@ -69,18 +66,19 @@ class FormConfigBuilder implements FormConfigBuilderInterface
      * @throws InvalidArgumentException if the data class is not a valid class or if
      *                                  the name contains invalid characters
      */
-    public function __construct(?string $name, ?string $dataClass, EventDispatcherInterface $dispatcher, array $options = [])
-    {
+    public function __construct(
+        ?string $name,
+        private ?string $dataClass,
```
Comment (xabbuh) [inline src/Symfony/Component/Form/FormConfigBuilder.php]: I would revert this change. In other places where we check the passed value to (for example) throw exceptions we don't use CPP.

File: src/Symfony/Component/Form/FormRegistry.php [hunk 6]
```diff
@@ -23,35 +23,28 @@
...
 
     /**
      * @param FormExtensionInterface[] $extensions
      *
      * @throws UnexpectedTypeException if any extension does not implement FormExtensionInterface
      */
-    public function __construct(array $extensions, ResolvedFormTypeFactoryInterface $resolvedTypeFactory)
-    {
+    public function __construct(
+        private array $extensions,
```
Comment (xabbuh) [inline src/Symfony/Component/Form/FormRegistry.php]: I would revert this change. In other places where we check the passed value to (for example) throw exceptions we don't use CPP.

File: src/Symfony/Component/Form/PreloadedExtension.php [hunk 7]
```diff
@@ -22,18 +22,19 @@ class PreloadedExtension implements FormExtensionInterface
 {
     private array $types = [];
     private array $typeExtensions = [];
```
Comment (xabbuh) [inline src/Symfony/Component/Form/PreloadedExtension.php]:
```suggestion
```

File: src/Symfony/Component/Form/PreloadedExtension.php [hunk 8]
```diff
@@ -22,18 +22,19 @@ class PreloadedExtension implements FormExtensionInterface
...
      * Creates a new preloaded extension.
      *
      * @param FormTypeInterface[]            $types          The types that the extension should support
      * @param FormTypeExtensionInterface[][] $typeExtensions The type extensions that the extension should support
      */
-    public function __construct(array $types, array $typeExtensions, FormTypeGuesserInterface $typeGuesser = null)
-    {
+    public function __construct(
+        array $types,
+        array $typeExtensions,
```
Comment (xabbuh) [inline src/Symfony/Component/Form/PreloadedExtension.php]:
```suggestion
        private array $typeExtensions,
```

File: src/Symfony/Component/Form/PreloadedExtension.php [hunk 9]
```diff
@@ -22,18 +22,19 @@ class PreloadedExtension implements FormExtensionInterface
...
+        private ?FormTypeGuesserInterface $typeGuesser = null,
+    ) {
         $this->typeExtensions = $typeExtensions;
```
Comment (xabbuh) [inline src/Symfony/Component/Form/PreloadedExtension.php]:
```suggestion
```

File: src/Symfony/Component/HttpClient/Response/TraceableResponse.php [hunk 10]
```diff
@@ -31,17 +31,16 @@
  */
 class TraceableResponse implements ResponseInterface, StreamableInterface
 {
-    private HttpClientInterface $client;
-    private ResponseInterface $response;
     private mixed $content;
```
Comment (xabbuh) [inline src/Symfony/Component/HttpClient/Response/TraceableResponse.php]:
```suggestion
```

File: src/Symfony/Component/HttpClient/Response/TraceableResponse.php [hunk 11]
```diff
@@ -31,17 +31,16 @@
...
-    private ?StopwatchEvent $event;
 
-    public function __construct(HttpClientInterface $client, ResponseInterface $response, &$content, StopwatchEvent $event = null)
+    public function __construct(
+        private HttpClientInterface $client,
+        private ResponseInterface $response,
+        &$content,
```
Comment (xabbuh) [inline src/Symfony/Component/HttpClient/Response/TraceableResponse.php]:
```suggestion
        private mixed &$content,
```

File: src/Symfony/Component/HttpClient/Response/TraceableResponse.php [hunk 12]
```diff
@@ -31,17 +31,16 @@
...
+        private ?StopwatchEvent $event = null,
+    )
     {
-        $this->client = $client;
-        $this->response = $response;
         $this->content = &$content;
```
Comment (xabbuh) [inline src/Symfony/Component/HttpClient/Response/TraceableResponse.php]:
```suggestion
```

File: src/Symfony/Component/Form/FormRegistry.php (hunk 6 above)
Comment (OskarStark) [reply src/Symfony/Component/Form/FormRegistry.php]: If I understood @nicolas-grekas correct, he explicitly wants that, but I am fine either way

File: src/Symfony/Component/HttpClient/Response/MockResponse.php (hunk 3 above)
Comment (OskarStark) [reply src/Symfony/Component/HttpClient/Response/MockResponse.php]: Indeed

File: src/Symfony/Component/HttpClient/Internal/AmpListener.php [hunk 13]
```diff
@@ -26,13 +26,15 @@
...
     /** @var resource|null */
     private $handle;
 
-    public function __construct(array &$info, array $pinSha256, \Closure $onProgress, &$handle)
-    {
+    public function __construct(
+        array &$info,
+        private array $pinSha256,
+        private \Closure $onProgress,
+        &$handle,
```
Comment (xabbuh) [inline src/Symfony/Component/HttpClient/Internal/AmpListener.php]:
```suggestion
        private &$handle,
```

File: src/Symfony/Component/Form/FormRegistry.php (hunk 6 above)
Comment (xabbuh) [reply src/Symfony/Component/Form/FormRegistry.php]: I'm fine either way, but we should then be consistent IMO

File: src/Symfony/Component/HttpClient/Internal/AmpListener.php (hunk 13 above)
Comment (OskarStark) [reply src/Symfony/Component/HttpClient/Internal/AmpListener.php]: I wasn't sure about this because of the reference

Comment (xabbuh) [reply src/Symfony/Component/HttpClient/Internal/AmpListener.php]: see https://3v4l.org/sQ1de

File: src/Symfony/Component/Form/Exception/TransformationFailedException.php (hunk 2 above)
Comment (OskarStark) [reply src/Symfony/Component/Form/Exception/TransformationFailedException.php]: cc @nicolas-grekas AFAIK you want to have a promoted property in those cases, right?

File: src/Symfony/Component/HttpClient/Internal/AmpListener.php (hunk 13 above)
Comment (OskarStark) [reply src/Symfony/Component/HttpClient/Internal/AmpListener.php]: Thanks! done

File: src/Symfony/Component/Form/FormRegistry.php (hunk 6 above)
Comment (nicolas-grekas) [reply src/Symfony/Component/Form/FormRegistry.php]: Can we use CPP in the classes @xabbuh mentions?
Also, please don't make pinging me an argument from authority, it's not and I don't want it to be ever. Better think and explain, and convince.

File: src/Symfony/Component/HttpClient/Retry/GenericRetryStrategy.php (hunk 4 above)
Comment (OskarStark) [reply src/Symfony/Component/HttpClient/Retry/GenericRetryStrategy.php]: Done

**Review by xabbuh — APPROVED** [review]

**Comment by xabbuh** [comment]
Thank you Oskar.
